types/index.d.ts

```diff
@@ -1000,6 +1001,11 @@ export declare namespace Knex {
     ): QueryBuilder<TRecord, TResult2>;
   }
 
+  interface OptimizerHint<TRecord extends {} = any, TResult extends {} = any> {
+    (...hints: string[]): QueryBuilder<TRecord, TResult>;
+    (hints: readonly string[]): QueryBuilder<TRecord, TResult>;
```

Comment:
I'm not sure if this typescript rule is correct. Please check.

Reply:
@lorefnon Could you take a look?